### ISSUE TITLE
Spurious MatchError warning in scala3doc

### ISSUE THREAD

**Issue by jphmrst** [lampepfl/dotty#12690]
## Compiler version

SBT 1.5.3, Scala 3.0.0

## Minimized code

```Scala
trait AAA[Y]
class BBB
def f[MM <: AAA[MM]](): BBB = new BBB
```

## Output

The compiler gives no warnings/errors, but the `doc` target in `sbt` (right out of sbt new scala/scala3.g8, no additions to `build.sbt`) gives a warning,

```scala
[warn] -- Warning: src/main/scala/Main.scala:4:4 --------------------------------------
[warn] 4 |def f[MM <: AAA[MM]](): BBB = new BBB
[warn]   |^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[warn]   |scala.MatchError: LazyRef(...) (of class dotty.tools.dotc.core.Types$LazyRef)
[warn]   |      at dotty.tools.scaladoc.tasty.TypesSupport.dotty$tools$scaladoc$tasty$TypesSupport$$inner(TypesSupport.scala:262)
[warn]   |      at dotty.tools.scaladoc.tasty.TypesSupport.inner$$anonfun$5(TypesSupport.scala:190)
[warn]   |      at scala.collection.immutable.List.map(List.scala:246)
[warn]   |      at dotty.tools.scaladoc.tasty.TypesSupport.dotty$tools$scaladoc$tasty$TypesSupport$$inner(TypesSupport.scala:190)
```

## Expectation

I'd expect no warnings here --- this seems like an internal matter.

**Comment by griggt** [comment]
Duplicate of #12327. Fixed in 3.0.1-RC1.

**Comment by som-snytt** [comment]
Per https://gitter.im/scala/scala?at=60b7cefb3d43a15a65de67fb
